test/test_cli.py

```diff
@@ -1,18 +1,39 @@
-from ansible_builder.cli import prepare
+from ansible_builder.cli import run, sys
+from contextlib import contextmanager
 
+import os
+import pytest
 
-def test_custom_image(exec_env_definition_file, tmpdir):
+from unittest import mock
+
+
+@pytest.fixture
+def cli_mock():
+    @contextmanager
+    def rf(args):
+        with mock.patch.object(sys, 'argv', args):
+            with pytest.raises(SystemExit) as exc:
+                yield exc
+    return rf
+
+
+def test_custom_image(exec_env_definition_file, tmpdir, cli_mock):
     content = {'version': 1}
     path = str(exec_env_definition_file(content=content))
 
-    aee = prepare(['build', '-f', path, '-b', 'my-custom-image', '-c', str(tmpdir)])
+    args = ['ansible-builder', 'build', '-f', path, '-b', 'my-custom-image', '-c', str(tmpdir)]
+    with cli_mock(args) as sys_exit:
+        run()
+
+    assert sys_exit.value.code == 0
 
-    assert aee.containerfile.base_image == 'my-custom-image'
+    with open(os.path.join(str(tmpdir), 'Containerfile')) as f:
+        assert 'FROM my-custom-image' in f.read()
 
 
-def test_build_context(good_exec_env_definition_path, tmpdir):
-    path = str(good_exec_env_definition_path)
-    build_context = str(tmpdir)
-    aee = prepare(['build', '-f', path, '-c', build_context])
+def test_invalid_path(tmpdir, cli_mock):
+    args = ['ansible-builder', 'build', '-f', 'foo/invalid', '-c', str(tmpdir)]
+    with cli_mock(args) as sys_exit:
+        run()
 
-    assert aee.build_context == build_context
```

Comment:
I'm a little concerned that this isn't actually testing anything now. I created the `prepare` function so that we could test that CLI args are parsed correctly and the appropriate attributes are set on the `AnsibleBuilder` object, without invoking the rest of the program. It seems like this is now just testing your mock. Am I missing something?

Reply:
But when you actually run the program, you don't run `prepare`, you run `run`. It doesn't matter if `prepare` parses the args correctly or not, because it has no bearing on what happens when the user it.

If you just need a helper methods for the test, then define `prepare` inside of this file.

This version is absolutely testing something. In fact, it's probably testing too much, becoming something more like an integration test with the only exception that it has the `autouse` mock of `run_command` in place.